File: pascals-triangle-ii/pascals-triangle-ii.cpp
class Solution {
public:
    vector<int> getRow(int rowIndex) {
        vector<vector<int>> m(rowIndex+1);
        for(int i=0;i<=rowIndex;i++){
            m[i].resize(i+1);
            m[i][0]=m[i][i]=1;
            for(int j=1;j<i;j++){
                m[i][j]=m[i-1][j-1]+m[i-1][j];
            }
        }
        return m[rowIndex];
    }
};

File: pascals-triangle-ii/pascals-triangle-ii.java
class Solution {
    public List<Integer> getRow(int rowIndex) {
        List<List<Integer>> triangle = new ArrayList<>();

        for(int i = 0; i <=rowIndex; i++){
            List<Integer> currRow = new ArrayList<>();
            for(int j = 0; j<=i; j++){
                if(j ==0 || j == i){
                    currRow.add(1);
                }else{
                    List<Integer> prevRow = triangle.get(i-1);
                    int sum = prevRow.get(j-1) + prevRow.get(j);
                    currRow.add(sum);
                }
            }
            triangle.add(currRow);
        }
        return triangle.get(rowIndex);
    }
}